### ISSUE TITLE
PnP JS - use in custom web application on client side

### ISSUE THREAD

**Issue by radukg** [pnp/pnpjs#1363]
Hi Team,

At the moment we have few SPFX customizations in our SharePoint Online tenant, and we are looking at options to move these pages to a custom web application (in Azure, URL will be like contoso.com). We are using pnpjs in order to access SharePoint content on behalf of current user, that is one of requirements of our security model. The question is - is there any way to setup pnpjs to authenticated current user in SPO or use access token provided by backend? If so, is there any CORS issues expected? 

Another option, as I see, is to introduce own backend API and start using it, but this option requires to update our client side code and replace pnpjs everywhere. 

### Category
- [ ] Enhancement
- [ ] Bug
- [X] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [2.0.6]

Please specify what version(s) of SharePoint you are targeting: [SPO]

### Expected / Desired Behavior / Question
I was expecting to found similar scenario in PnPJS documentation

### Observed Behavior
I found solutions how to use PnPJS in NodeJS application, but it is not the same as using it in browser (due to CORS).

**Comment by koltyakov** [comment]
[ADAL](https://pnp.github.io/pnpjs/adaljsclient/adalclient/#setup-and-use-with-microsoft-graph) (the sample shows Graph, but consuming SP API from an external resource is similar) or [MSAL](https://spblog.net/post/2019/06/04/building-single-page-application-with-react-msal-js-and-pnpjs) is the way to go.

**Comment by radukg** [comment]
Thank you for quick response!

As I understood MSAL in PnPJS could be used only in one way - with browser popups (correct me if I am wrong). Most of our clients have blocked popups (default browser behaviour), is there any way to configure MSAL in PnPjs (MsalClientSetup) to use redirect mode? As I mentioned we have backend and we can handle redirect requests from Azure AD and provide access tokens to PnPJS on client side

**Comment by koltyakov** [comment]
There should be no or little issues related to CORS with SPO. I personally have a couple of projects with such a model which deal with SharePoint API for document management and a variety of other APIs and no CORS-related problems.

The only issues I faced were related to IE. Because of which back in days I got to fork and rewrite ADAL.js from the grounds up to support the app case, as it has some limitation being embedded to a 3rd party and not allowing redirects. But it's completely a different story and luckily IE is almost not an option anymore even for some hardcore enterprise customers.
